src/application/hooks/api/collection/index.ts

```diff
@@ -0,0 +1,90 @@
+import type { UseQueryOptions } from "@tanstack/react-query";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+import { api } from "@/infra/api";
+import type { GetCollectionCheckResponse } from "@/infra/api/collection/types";
+
+import { useSuspendedQuery } from "../core";
+import { QUERY_KEYS } from "./queryKey";
+
+/**
+ * NOTE
+ * 임시: 밈별 콜렉션 정보 API
+ * 백엔드에 밈별 콜렉션 정보 API 인터페이스 요청하기
+ */
+export const useGetCollectionCheck = <T = GetCollectionCheckResponse>(
+  memeId: number,
+  options?: UseQueryOptions<GetCollectionCheckResponse, any, T>,
+) => {
+  return useSuspendedQuery({
+    queryKey: QUERY_KEYS.getCollectionCheck(memeId),
+    queryFn: () => api.collection.getCollectionCheck(memeId),
+    ...options,
+  });
+};
+
+/**
+ * 콜렉션 삭제 API
+ */
+export const useDeleteMemeFromCollection = ({ memeId }: { memeId: number }) => {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: () => api.collection.deleteMemeFromCollection(memeId),
+    onMutate: async (deletedMemeId: number) => {
```

Comment:
```suggestion
export const useDeleteMemeFromCollection = () => {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: api.collection.deleteMemeFromCollection,
    onMutate: async (deletedMemeId) => { ... }
   ...
   // 사용 : mutate(deleteId)
```
- `mutate()` 할때 ID를 전달해줄 수 있어요!

장점 : api 함수를 그대로 mutateFn에 전달하면 타입을 덜써도 됨
- 커스텀 mutation hook 인자
-  onMutate 인자 (자동으로 타입추론 됨)